src/subsplit_dag.cpp

```diff
@@ -466,29 +486,34 @@ void SubsplitDAG::BuildNodesDepthFirst(const SizeBitsetMap &index_to_child,
   CreateAndInsertNode(subsplit);
 }
 
-void SubsplitDAG::BuildNodes(const SizeBitsetMap &index_to_child) {
+void SubsplitDAG::BuildNodes(const SizeBitsetMap &index_to_child,
+                             const BitsetVector &rootsplits) {
   std::unordered_set<Bitset> visited_subsplits;
 
   // We will create fake subsplits and insert to dag_nodes_.
   // These nodes will take IDs in [0, taxon_count_).
-  Bitset zero(taxon_count_);
   for (size_t taxon_idx = 0; taxon_idx < taxon_count_; taxon_idx++) {
     CreateAndInsertNode(
         Bitset::FakeSubsplit(Bitset::Singleton(taxon_count_, taxon_idx)));
   }
   // We are going to add the remaining nodes.
   // The root splits will take on the higher IDs compared to the non-rootsplits.
-  for (const auto &rootsplit : rootsplits_) {
+  for (const auto &rootsplit : rootsplits) {
     const auto subsplit = rootsplit + ~rootsplit;
     BuildNodesDepthFirst(index_to_child, subsplit, visited_subsplits);
   }
+  // We will create a root node with a bitset where the first half is all zeros and the
```

Comment:
This is great documentation but I think it deserves slightly higher prominence, say in the header file.